src/sys/unix.rs

```diff
@@ -1406,6 +1406,62 @@ pub(crate) const fn to_mreqn(
     }
 }
 
+/// Get the value for the `SO_ORIGINAL_DST` option on this socket.
+///
+/// This value contains the original destination IPv4 address of the connection
+/// redirected using `iptables` `REDIRECT` or `TPROXY`.
```

Comment:
Nit: all these docs are now hidden, so we might as well remove them or move the to the public method on `Socket`, same for the Windows side.
```suggestion
```